### ISSUE TITLE
Client Crash / Freeze

### ISSUE THREAD

**Issue by rubybrowncoat** [CoFH/Feedback#638]
Multiplayer. Game loads normally, and I can join the server.

The crash OR freeze (it's quite random how it manifests) can happen 20 seconds after joining the server, 20 minutes or never. It is unpredictable and I haven't been able to reproduce it.

The beginning of the crash report states this:
```
java.lang.NullPointerException: Unexpected error
	at cofh.lib.util.LinkedHashList.delete(LinkedHashList.java:267)
	at cofh.lib.util.LinkedHashList.shift(LinkedHashList.java:168)
	at net.minecraft.world.World.func_72939_s(World.java:50532)
	at net.minecraft.client.Minecraft.func_71407_l(Minecraft.java:2006)
	at net.minecraft.client.Minecraft.func_71411_J(Minecraft.java:973)
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:898)
	at net.minecraft.client.main.Main.main(SourceFile:148)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)
```

[The rest of the crash log](http://pastebin.com/VMrVnqeX)

It is happening in the Agrarian Skies 2 1.1.6 modpack. I have already reported it on AS2's curseforge page, but I thought I'd report the problem here too.

I have tried fiddling with all graphic options thinking there may be a problem there but it happens regardless.

Additionally, it has happened in all sorts of situations. It isn't specific to any action or surrounding that I can isolate. I had it happen while punching trees, hovering over items, walking over slabs, opening inventories or other interfaces, falling through the void, jumping, and several more places and moments.

It appears to be completely random, but consistent in its persistence. It just keeps happening.

**Comment by KingLemming** [comment]
Alright, so here's the thing.

We're replacing Mojang's unoptimized list with a version of our own. That's it. This error is fundamentally occurring elsewhere, and would have occurred regardless of which list in place.

There is something deeply wrong with your install or setup, potentially a hardware error. If it were in the list code, it would happen absolutely immediately every time anyone started a game.

Basically, we're providing a car, and somebody is putting a bomb in that car, and the log only shows the car exploding.

**Comment by rubybrowncoat** [comment]
The install is Agrarian Skies 2 1.1.6 from the Curse Voice launcher. I will try a manual install and see if that changes anything but honestly I don't believe it will as I have already checked all the files.

Thanks for the reply, anyway. I will report here if anything changes.

**Comment by KingLemming** [comment]
Yeah, that's why I suggested a hardware error. If your RAM is just failing and losing data, that could cause it.

**Comment by XFactHD** [comment]
I can prove that this is not a bug in CoFHCore/CoFHLib. I got the same console output some time ago every time I came near Mekanism solar evaporation plants too quickly. Aiden looked into it and it got solved. This means that your problem is definitely caused by another mod.

**Comment by rubybrowncoat** [comment]
I figured it wasn't a problem directly with CoFH but I though it may be something that needed a little attention. But things got a little weirder.

Here's what I did just now.

I was editing a bad tileentity in MCEdit to fix a server crash when I noticed a lot of forge microblocks that I had mined out weeks before. They do not appear in game. They are... nothing. They are invisible and untouchable in the loaded world. Yet they were showing up in MCEdit, somehow. 

That was strange so I manually deleted them and I haven't had a crash/freeze in the past hour.

I can only presume these "ghost" microblocks may be the cause of this instability.

I will test the waters more tomorrow.
